suggester/src/main/java/org/opengrok/suggest/popular/impl/chronicle/ChronicleMapAdapter.java

```diff
@@ -120,24 +120,29 @@ public void resize(final int newMapSize, final double newMapAvgKey) throws IOExc
 
         Path tempFile = Files.createTempFile("opengrok", "chronicle");
 
-        map.getAll(tempFile.toFile());
+        ChronicleMap<BytesRef, Integer> m;
 
-        String field = map.name();
+        try {
+            map.getAll(tempFile.toFile());
 
-        map.close();
+            String field = map.name();
 
-        Files.delete(f.toPath());
+            map.close();
 
-        ChronicleMap<BytesRef, Integer> m = ChronicleMap.of(BytesRef.class, Integer.class)
-                .name(field)
-                .averageKeySize(newMapAvgKey)
-                .entries(newMapSize)
-                .keyReaderAndDataAccess(BytesRefSizedReader.INSTANCE, new BytesRefDataAccess())
-                .createOrRecoverPersistedTo(f);
+            Files.delete(f.toPath());
 
-        m.putAll(tempFile.toFile());
-
-        Files.delete(tempFile);
+            m = ChronicleMap.of(BytesRef.class, Integer.class)
+                    .name(field)
+                    .averageKeySize(newMapAvgKey)
+                    .entries(newMapSize)
+                    .keyReaderAndDataAccess(BytesRefSizedReader.INSTANCE, new BytesRefDataAccess())
+                    .createOrRecoverPersistedTo(f);
+            m.putAll(tempFile.toFile());
+        } catch (IOException e) {
+            throw e;
+        } finally {
+            Files.delete(tempFile);
+        }
 
         map = m;
```

Comment:
I would move this assignment inside the `try` block. It makes it clear that this will be only done if no exception is thrown.